python/ray/train/_internal/worker_group.py

```diff
@@ -281,16 +290,22 @@ def execute_single_async(
                 f"The provided worker_index {worker_index} is "
                 f"not valid for {self.num_workers} workers."
             )
-        return self.workers[worker_index].actor._RayTrainWorker__execute.remote(
-            func, *args, **kwargs
+        return (
+            self.workers[worker_index]
+            .actor._RayTrainWorker__execute.options(
+                name=f"_RayTrainWorker__execute.{name}"
+            )
+            .remote(func, *args, **kwargs)
         )
 
     def execute_single(
-        self, worker_index: int, func: Callable[..., T], *args, **kwargs
+        self, name: str, worker_index: int, func: Callable[..., T], *args, **kwargs
     ) -> T:
         """Execute ``func`` on worker with index ``worker_index``.
 
         Args:
+            name: Allows customized task name to be used rather than
```

Comment:
can we just extract this from `func.__name__` instead of requiring it as an argument?

Reply:
sounds good. I was originally thinking what if user just provides a `def f()`, and we will still lose information about what this is...
But I guess this API is largely internal ATM so I am gonna change to what you suggest!